core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2178,23 +2180,24 @@ class WorkspaceService(protected val userInfo: UserInfo,
     )
   }
 
-  private def syncPolicies(workspaceId: String, policyEmailsByName: Map[SamResourcePolicyName, WorkbenchEmail], workspaceRequest: WorkspaceRequest, parentSpan: Span) = {
-    traceWithParent("traversePolicies", parentSpan) (s1 =>
-      Future.traverse(policyEmailsByName.keys) { policyName =>
-        if (policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
+  private def syncPolicies(workspaceId: String, policyEmailsByName: Map[SamResourcePolicyName, WorkbenchEmail], workspaceRequest: WorkspaceRequest, parentSpan: Span): Future[Unit] = {
+    traceWithParent("traversePolicies", parentSpan) { s1 =>
+      // This used to be Future.traverse but that runs all the futures eagerly and concurrently. This led to a perf
+      // issue with hitting the akka subscription timeout. We switched to use cats traverse to make these calls serially.
+      policyEmailsByName.keys.toList.traverse_ { policyName =>
+        Applicative[IO].whenA(
           // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
           // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
           // the limit of 2000
-          Future.successful(())
-        } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined || policyName == SamWorkspacePolicyNames.canCompute) {
           // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
           // granted bucket access (and thus need a google group)
-          traceWithParent(s"syncPolicy-${policyName}", s1)(_ => samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspaceId, policyName))
-        } else {
-          Future.successful(())
-        }
-      }
-    )
+          !(policyName.isProjectOwner && !workspaceRequest.hasAuthorizationDomain) &&
+            (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined || policyName.isCanCompute)) (
```

Comment:
I'm not thrilled about this logic statement, it took me a long time to grok what this is trying to say which has me worried if we have it right in the first place and has me worried that this will be fragile/difficult to maintain.

Long story short, I wonder if we create a new enumeration/list of policy names that should be synced.  Something like: 

`// Seq[String] of the policy names on a Workspace that need to be sync'd with their corresponding policy group on GSuite
val policyNamesThatRequireSync: Seq[String] = WorkspaceAccessLevels.all.map(toString) :+ SamWorkspacePolicyNames.canCompute.toString()`

(might be able to pretty that up too)

And then if we apply DeMorgan's to the first predicate, I think the logic can be expressed more simply as:

`(workspaceRequest.hasAuthorizationDomain || !policyName.isProjectOwner) && policyNamesThatRequireSync.contains(policyName.value)`

Which I think boils down do the idea that if the policy is one a "sync'able" policy and the Workspace has an AuthDomain, then go ahead and sync it.  If the Workspace does not have an AuthDomain, then skip the `Project Owner` policy.  Which has me thinking instead we go to:

`// Seq[String] of the policy names on a Workspace that need to be sync'd with their corresponding policy group on GSuite.  If there is not an AuthDomain, then we do not need to sync the Project Owner policy
def policyNamesThatRequireSync(withAuthDomain: Boolean): Seq[String] = { 
    val policyNames = WorkspaceAccessLevels.all.map(toString) :+ SamWorkspacePolicyNames.canCompute.toString()
    if (withAuthDomain) policyNames else policyNames.filterNot(_ == WorkspaceAccessLevels.ProjectOwner.value)
}`

So finally, I think this logical expression I'm commenting on can be boiled down to:

`policyNamesThatRequireSync(workspaceRequest.hasAuthorizationDomain).contains(policyName.value)`

Sorry this is such a long comment, it took me several tries to think this all through and to try to express this logic more clearly.  I really really hope I've got it right, I'm not convinced I do, which is exactly the reason why I wanted to try to simplify the expression.